akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -204,6 +204,17 @@ object Source {
     })
 
   /**
+   * Create [[Source]] that will continually produce given elements in specified order.
+   *
+   * Start a new 'cycled' `Source` from the given elements. The producer stream of elements
+   * will continue infinitely by repeating the same sequence of elements provided during construction.
+   */
+  def cycle[T](f: () ⇒ Iterator[T]): Source[T, NotUsed] = fromIterator(() ⇒ {
+    if (f().isEmpty) Iterator.empty
```

Comment:
ok, I guess it's a matter of how we define it.

Reference docs and scaladocs should be updated regarding empty iterator.

Do we want `f` to be evaluated for each round or only once? Now it looks like a mix, it is checked for empty once but called each time by `Iterator.continually`.